### PR TITLE
Header hamburger

### PR DESCRIPTION
**Pull request by thasnim-cm**

### CODE REVIEW

File: src/components/appcontent.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                                {openSearchField &&
+                                    <SearchField searchValue={searchValue} setSearchValue={setSearchValue}
+                                    />
+                                }
+                                {appContent}
+                            </div>
+                        </div>
+                    }
+
+                    <div className='hidden sm:block'>
```
Comment (raj-kumar-litmus) [inline src/components/appcontent.tsx]: 1. Will this be always hidden ?
2. What is usually present in appContent (string or something else ?)

File: src/App.tsx [hunk 2]
```diff
@@ -1,4 +1,12 @@
-function App() {
-  return <div></div>;
+import AppContent from "./components/appcontent";
+
+function App ({ appContent }: { appContent: any }) {
+  return <div>
+    <AppContent
+      showSidePane
+      showNavbar
```
Comment (raj-kumar-litmus) [inline src/App.tsx:12]: is Showsidepane and shownavbar always true ?

File: src/components/appcontent.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import DataUsageIcon from '../assets/data_usage.svg';
+import DCOpenOrdersIcon from '../assets/dc_open_orders.svg';
+import LeaderboardIcon from '../assets/leaderboard.svg';
+import LegendToggleIcon from '../assets/legend_toggle.svg';
+import Navbar from './common/navbar';
+import SearchField from './common/searchfield';
+import SidePaneGrid from './common/sidepanegrid';
+import SidePaneList from './common/sidepanelist';
+
+const MENU_LIST = [
```
Comment (raj-kumar-litmus) [inline src/components/appcontent.tsx]: Keep this list in a utils file so that it is easy to configure in the future

File: src/App.tsx [hunk 4]
```diff
@@ -1,4 +1,12 @@
-function App() {
-  return <div></div>;
+import AppContent from "./components/appcontent";
+
+function App ({ appContent }: { appContent: any }) {
```
Comment (raj-kumar-litmus) [inline src/App.tsx]: instead of comparing proptypes like this, do it the way you have done in appContext.tsx

File: src/components/appcontent.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+type Props = {
+    showSidePane: boolean;
+    showNavbar: boolean;
+    appContent: any;
+};
+
+const AppContent: FC<Props> = ({ showSidePane, showNavbar, appContent }) => {
+    const [showSidePaneGrid, setShowSidePaneGrid] = useState<boolean>(false);
+    const [openSearchField, setOpenSearchField] = useState<boolean>(false);
```
Comment (raj-kumar-litmus) [inline src/components/appcontent.tsx:16]: Why not move this line into NavBar component itself.

File: src/components/appcontent.tsx (hunk 1 above)
Comment (thasnim-cm) [reply src/components/appcontent.tsx]: 1. That's for the web screen and will remain hidden on mobile screens.
2. The entire page excluding header and sidepane

File: src/components/appcontent.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+                <div className='flex'>
+                    {showSidePane &&
+                        <SidePaneList
+                            menuList={MENU_LIST}
+                            selectedMenu={selectedMenu}
+                            setSelectedMenu={setSelectedMenu}
+                        />
+                    }
+                    {showSidePane &&
+                        <div className={`${showSidePaneGrid ? `bg-zinc-400` : ''} flex w-full block sm:hidden`} onClick={() => setShowSidePaneGrid(false)}>
```
Comment (raj-kumar-litmus) [inline src/components/appcontent.tsx:40]: `sm:block`, `sm:hidden`
Where are these classes defined ?

File: src/App.tsx (hunk 2 above)
Comment (thasnim-cm) [reply src/App.tsx:12]: No. It depends on whether to show it or not. We can set to false to hide navbar or sidepane

File: src/components/common/searchbar.tsx [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    setOpenSearchField: (a: boolean) => void;
+    searchValue: string;
+    setSearchValue: (a: string) => void;
+};
+
+const SearchBar: FC<Props> = ({ showSearchButton, setOpenSearchField, openSearchField, searchValue, setSearchValue }) => {
+
+    return (
+        <div className="block">
+            <div className='flex'>
```
Comment (raj-kumar-litmus) [inline src/components/common/searchbar.tsx]: Lines 17 and 18 can be merged in to a single div.

File: src/components/common/searchfield.tsx [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    searchValue: string;
+    setSearchValue: (a: string) => void;
+};
+
+const SearchField: FC<Props> = ({ searchValue, setSearchValue }) => {
+
+    return (
+        <div className="flex sm:hidden px-6 py-3 border-slate-200  border-solid	 border-b justify-between">
+            <InputText type="text" name="q" value={searchValue} onChange={(e) => setSearchValue(e.target.value)} className="w-full focus:outline-none placeholder:text-stone-500 bg-transparent placeholder:font-helvetica placeholder:font-medium" placeholder="Search Order" />
+            <button type="submit" className='-ml-8'>
```
Comment (raj-kumar-litmus) [inline src/components/common/searchfield.tsx]: Use prime react buttons to maintain uniformity across the project.

File: src/components/common/searchfield.tsx [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+                <img className="h-[13px] rotate-180" src={RightArrowIcon} alt="Search" />
```
Comment (raj-kumar-litmus) [inline src/components/common/searchfield.tsx]: you have used prime react image in other places. 
use the same here.

File: src/components/common/sidepanegrid.tsx [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+    menuList: Array<IMenu>;
+    selectedMenu: number;
+    setSelectedMenu: (a: number) => void;
+};
+
+const SidePaneGrid: FC<Props> = ({ menuList, selectedMenu, setSelectedMenu }) => {
+
+    return (
+        <div className='transition-all ease-in-out duration-100 max-w-[350px]  border-slate-200 shadow-slate-200 border-solid border-r min-h-[calc(100vh-2.6rem)] z-10 bg-white' >
+            <div className='flex flex-wrap p-1'>
```
Comment (raj-kumar-litmus) [inline src/components/common/sidepanegrid.tsx:15]: merge divs in lines 14 and 15

File: src/components/common/sidepanelist.tsx [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    menuList: Array<IMenu>;
+    selectedMenu: number;
+    setSelectedMenu: (a: number) => void;
+};
+
+const SidePaneList: FC<Props> = ({ menuList, selectedMenu, setSelectedMenu }) => {
+
+    return (
+        <div className='hidden sm:block w-80 border-slate-200 shadow-slate-200 border-solid border-r min-h-[calc(100vh-4.6rem)]'>
+            <div>
```
Comment (raj-kumar-litmus) [inline src/components/common/sidepanelist.tsx]: This div is not needed

File: src/components/appcontent.tsx (hunk 5 above)
Comment (thasnim-cm) [reply src/components/appcontent.tsx:16]: These states are used to apply some styles and display search component in appcontent.tsx

File: src/components/appcontent.tsx (hunk 3 above)
Comment (thasnim-cm) [reply src/components/appcontent.tsx]: ok

File: src/App.tsx (hunk 2 above)
Comment (raj-kumar-litmus) [reply src/App.tsx:12]: `We can set to false`.
In this exisiting code, it is always set to true.
Right ?
in the case, why are we passing it like this ?

File: src/components/appcontent.tsx (hunk 6 above)
Comment (raj-kumar-litmus) [reply src/components/appcontent.tsx:40]: This is not yet addressed.

File: src/components/common/searchbar.tsx (hunk 7 above)
Comment (raj-kumar-litmus) [reply src/components/common/searchbar.tsx]: This is not yet addressed.

File: src/components/common/searchfield.tsx (hunk 8 above)
Comment (raj-kumar-litmus) [reply src/components/common/searchfield.tsx]: This is not yet addressed.

File: src/components/common/searchfield.tsx (hunk 9 above)
Comment (raj-kumar-litmus) [reply src/components/common/searchfield.tsx]: This is not yet addressed.

File: src/components/common/sidepanegrid.tsx (hunk 10 above)
Comment (raj-kumar-litmus) [reply src/components/common/sidepanegrid.tsx:15]: This is not yet addressed.

File: src/components/common/sidepanelist.tsx (hunk 11 above)
Comment (raj-kumar-litmus) [reply src/components/common/sidepanelist.tsx]: This is not yet addressed.

File: src/App.tsx (hunk 2 above)
Comment (thasnim-cm) [reply src/App.tsx:12]: We can pass the page inside AppContent component's appContent prop.. And the showSidePane and showNavbar can be set as per the requirement

File: src/components/appcontent.tsx (hunk 6 above)
Comment (thasnim-cm) [reply src/components/appcontent.tsx:40]: These are tailwind's inbuild classes.. the styles are shown after checking the state value of showSidePaneGrid

File: src/components/common/sidepanegrid.tsx (hunk 10 above)
Comment (thasnim-cm) [reply src/components/common/sidepanegrid.tsx:15]: Some styling issues are coming after merging those divs

File: src/components/common/sidepanelist.tsx (hunk 11 above)
Comment (thasnim-cm) [reply src/components/common/sidepanelist.tsx]: Will remove and push the code

File: src/components/common/searchbar.tsx (hunk 7 above)
Comment (thasnim-cm) [reply src/components/common/searchbar.tsx]: Merged them

Comment (raj-kumar-litmus) [reply src/components/common/searchbar.tsx]: I see that you removed the flex class.
Did u check to see if the UI got affected after removing ?

File: src/components/common/sidepanegrid.tsx (hunk 10 above)
Comment (raj-kumar-litmus) [reply src/components/common/sidepanegrid.tsx:15]: Try to address the issue. If you are stuck, let me know.

File: src/components/common/searchfield.tsx (hunk 9 above)
Comment (thasnim-cm) [reply src/components/common/searchfield.tsx]: Pushed the changes

File: src/components/common/searchfield.tsx (hunk 8 above)
Comment (thasnim-cm) [reply src/components/common/searchfield.tsx]: Pushed the changes

File: src/components/common/searchbar.tsx (hunk 7 above)
Comment (thasnim-cm) [reply src/components/common/searchbar.tsx]: I checked. It works fine

File: src/components/common/searchfield.tsx [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+    searchValue: string;
+    setSearchValue: (a: string) => void;
+};
+
+const SearchField: FC<Props> = ({ searchValue, setSearchValue }) => {
+
+    return (
+        <div className="flex sm:hidden px-6 py-3 border-slate-200  border-solid	 border-b justify-between">
+            <InputText type="text" name="q" value={searchValue} onChange={(e) => setSearchValue(e.target.value)} className="w-full focus:outline-none placeholder:text-stone-500 bg-transparent placeholder:font-helvetica placeholder:font-medium" placeholder="Search Order" />
+            <Button type="submit" className='-ml-8'>
```
Comment (raj-kumar-litmus) [inline src/components/common/searchfield.tsx:17]: `-ml-8`
I dont think this is the correct class name.
Please check once.

File: src/components/common/searchfield.tsx [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+                <Image className="arrow-icon" src={RightArrowIcon} alt="Search" />
+            </Button>
```
Comment (raj-kumar-litmus) [inline src/components/common/searchfield.tsx:19]: Also, please create re-usable components for Image and Button and use the components in all places.
Create storybook for those button and Image components.

File: src/components/common/searchfield.tsx (hunk 12 above)
Comment (thasnim-cm) [reply src/components/common/searchfield.tsx:17]: Its margin-left:-2rem

File: src/components/common/searchfield.tsx (hunk 13 above)
Comment (thasnim-cm) [reply src/components/common/searchfield.tsx:19]: ok

File: src/components/common/searchbar.tsx [hunk 14]
```diff
@@ -0,0 +1,29 @@
+import { FC } from 'react';
+import SearchIcon from "../../assets/search.svg";
+import { InputText } from "primereact/inputtext";
+import { Button } from 'primereact/button';
```
Comment (raj-kumar-litmus) [inline src/components/common/searchbar.tsx:3]: InputText should be a reusable component.
Also use the reusasble button component here.

**Pull request merged**